OTVision/detect/cli.py

```diff
@@ -0,0 +1,173 @@
+        )
+        self._parser.add_argument(
+            "--log_level_console",
+            type=str,
+            choices=VALID_LOG_LEVELS,
+            help="Log level for logging to the console",
+            required=False,
+        )
+        self._parser.add_argument(
+            "--log_level_file",
+            type=str,
+            choices=VALID_LOG_LEVELS,
+            help="Log level for logging to a log file",
+            required=False,
+        )
+        self._parser.add_argument(
+            "--logfile",
+            default=str(DEFAULT_LOG_FILE),
+            type=str,
+            help="Specify log file directory.",
+            required=False,
+        )
+        self._parser.add_argument(
+            "--logfile_overwrite",
+            action="store_true",
+            help="Overwrite log file if it already exists.",
+            required=False,
+        )
+        self._parser.add_argument(
+            "--detect_start",
+            default=None,
+            type=int,
+            help="Specify start of detection in seconds.",
+            required=False,
+        )
+        self._parser.add_argument(
+            "--detect_end",
+            default=None,
+            type=int,
+            help="Specify end of detection in seconds.",
+            required=False,
+        )
+
+    def parse(self) -> DetectCliArgs:
+        args = self._parser.parse_args(self._argv)
+        self.__assert_cli_args_valid(args)
+
+        return DetectCliArgs(
+            paths=self._parse_files(args.paths),
+            config_file=args.config,
+            weights=args.weights,
+            conf=float(args.conf) if args.conf is not None else None,
+            iou=float(args.iou) if args.iou is not None else None,
+            imagesize=int(args.imagesize) if args.imagesize is not None else None,
+            expected_duration=(
+                timedelta(seconds=args.expected_duration)
+                if args.expected_duration is not None
+                else None
+            ),
+            half=bool(args.half) if args.half else None,
+            overwrite=args.overwrite,
+            detect_start=(
+                int(args.detect_start) if args.detect_start is not None else None
+            ),
+            detect_end=int(args.detect_end) if args.detect_end is not None else None,
+            logfile=Path(args.logfile),
+            log_level_console=args.log_level_console,
+            log_level_file=args.log_level_file,
+            logfile_overwrite=args.logfile_overwrite,
+        )
+
+    def __assert_cli_args_valid(self, args: Namespace) -> None:
+        if args.paths is None and args.config is None:
+            raise CliParseError(
+                (
+                    "No paths have been passed as command line args."
+                    "No paths have been defined in the user config."
+                )
+            )
+        if args.expected_duration is None:
```

Comment:
This check must happen to a later point. The expected_duration might be specified in the config.yaml file.

Reply:
Actually we can get rid of this check entirely since expected_duration is now optional. The actual duration is now used in case expected_duration is not specified.